### PR TITLE
Update to 1.20.6

### PR DESCRIPTION
**Pull request by Patbox**
This pr updates trinkets to version 1.20.6, with minimal api changes to make it match with how MC handles things (mostly moving from EntityAttribute to RegistryEntry<>, as MC requires them everywhere) and to update the dependencies.

Additionally it adds basic support for datafixers updating ItemStacks stored in trinket's inventory (this is unlikely to require any changes in the future).

Solves #307

### CODE REVIEW

**Review by emilyploszaj — CHANGES_REQUESTED** [review]
The codebase doesn't use star imports and uses tabs instead of spaces. At least one large file has replaced all tabs with spaces making the diff completely illegible, and the added files use spaces instead of tabs, this will need to be resolved before being properly reviewed.

**Comment by Patbox** [comment]
Done

**Comment by HumanGamer** [comment]
I just tried this 1.20.6 version and noticed an issue. With keep inventory enabled, on death, trinkets get deleted. I tried to debug this but couldn't figure out why. Not sure if this is an issue outside of this PR but I've only tried on 1.20.6 with this.

**Comment by Patbox** [comment]
Should be fixed now

File: src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java
Comment (emilyploszaj) [inline src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java:56]: SuppressWarnings also unneeded here.

File: src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java
Comment (emilyploszaj) [inline src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java:171]: Don't use `var`.

File: src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java
Comment (emilyploszaj) [inline src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java:235]: Not that important but why the change from a string to a char here? I preferred the string.

File: src/main/java/dev/emi/trinkets/mixin/datafixer/Schema1460Mixin.java
Comment (emilyploszaj) [inline src/main/java/dev/emi/trinkets/mixin/datafixer/Schema1460Mixin.java:19]: I said it in discord but I do not care for this, I can't easily understand this part of the codebase, all of the methods are unmapped, and the benefit is pretty low level that I've never supported before, I'd prefer this class be removed.

File: src/main/resources/trinkets.mixins.json
Comment (emilyploszaj) [inline src/main/resources/trinkets.mixins.json:11]: Don't adjust the order of the mixins here, subpackages (`accessor`) go first.

**Review by emilyploszaj — CHANGES_REQUESTED** [review]
Functionality wise I imagine it works and looks good in that sense. There were a lot of crits here, mostly style stuff, I don't know if I caught them all but my codebases don't use star imports or `var`, `info` is always used instead of `ci` or `cir`, and the handful of spot specific things I mentioned. Some comments need further clarification.

File: src/main/java/dev/emi/trinkets/api/TrinketsAttributeModifiersComponent.java
Comment (Patbox) [reply src/main/java/dev/emi/trinkets/api/TrinketsAttributeModifiersComponent.java:20]: It replaces reading attributes from nbt, as ItemStacks use components now (also as a bonus, it works like an api extension as it can be set on items by default)

File: src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java
Comment (Patbox) [reply src/main/java/dev/emi/trinkets/mixin/LivingEntityMixin.java:56]: Added it purely for dev, as my ide marked all code post casting as unreachable, but can remove that

File: src/main/resources/trinkets.mixins.json
Comment (Patbox) [reply src/main/resources/trinkets.mixins.json:11]: Going to blame the ide here, but going to fix that

File: src/main/java/dev/emi/trinkets/mixin/datafixer/Schema1460Mixin.java
Comment (Patbox) [reply src/main/java/dev/emi/trinkets/mixin/datafixer/Schema1460Mixin.java:19]: I, as a developer/user of this library prefer for Trinkets to support this, as losing things like item data/enchantments isn't fun. In cases where people use trinkets purely for things as elytra cape slot will likely want to not lose the data.

The only thing happening here is adding nbt structure of trinkets data to definitions of player and entity. It doesn't do anything else, as vanilla handles the rest. These methods are unnamed, as they are just lambdas, which never have a name in yarn.

File: src/main/java/dev/emi/trinkets/api/Trinket.java
Comment (Patbox) [reply src/main/java/dev/emi/trinkets/api/Trinket.java:134]: Yes it got removed

File: src/main/java/dev/emi/trinkets/api/Trinket.java
Comment (Patbox) [reply src/main/java/dev/emi/trinkets/api/Trinket.java:119]: Mostly written that on a laptop and it just didn't fit on the screen. But cqn fix that

File: src/main/java/dev/emi/trinkets/mixin/ClientPlayNetworkHandlerMixin.java
Comment (Patbox) [reply src/main/java/dev/emi/trinkets/mixin/ClientPlayNetworkHandlerMixin.java:29]: Mostly because it was done by ide instead of by hand, through going to replace it with @Local from MixinExtras instead

**Comment by Patbox** [comment]
Fixed all the style issues.

I'm keeping datafixer, but commented on what every part of it does

**Comment by Lakuna** [comment]
Hello @Patbox,

I updated one of my mods to 1.20.6 using this fork and found that Trinkets is now required on the client side. This appears to be related to Ladysnake/Cardinal-Components-API#174, which is addressed in the migration guide [here](https://ladysnake.org/wiki/cardinal-components-api/upgrade-instructions/CCA-6-changes#client-optional-components). It would be nice if this could be fixed before this fork is merged (it appears to work fine otherwise).

Thanks!

**Comment by Patbox** [comment]
@Lakuna Could you tell me which case you would need that? Trinkets (at least the original version) never really worked with vanilla/non-trinkets environments, as it's not possible to interact with these slots at all. It can also be confusing, as trinkets changes shift clicking behaviour to put items within it's slot to add support for it. Additionally it wouldn't work anyway, as TrinketsAttributeModifiersComponent, which was added to replicate attribute support in nbt, but also can be used for defining the attributes in Item.Settings is required to be on both sides

**Comment by Lakuna** [comment]
> @Lakuna Could you tell me which case you would need that? Trinkets (at least the original version) never really worked with vanilla/non-trinkets environments, as it's not possible to interact with these slots at all. It can also be confusing, as trinkets changes shift clicking behaviour to put items within it's slot to add support for it. Additionally it wouldn't work anyway, as TrinketsAttributeModifiersComponent, which was added to replicate attribute support in nbt, but also can be used for defining the attributes in Item.Settings is required to be on both sides

My server was previously able to be connected to by vanilla clients who didn't wish to use trinkets or by clients with the mod who did want to use trinkets. The only mod that I use that "adds" a trinket is one that allows Elytra to be used in the cape slot. Since this update, players that belong to the former group are no longer able to connect. It would be strange to me if this update would impose a restriction that didn't exist beforehand.

**Comment by Patbox** [comment]
Going to leave that decision for @emilyploszaj . For myself I will update the polymer port for server side usage

**Comment by rakantoh** [comment]
sorry if i an noob, but where i can  download the commits/forks made by @Patbox? Thanks.

**Review by rakantoh — APPROVED** [review]

**Review by emilyploszaj — APPROVED** [review]
Docs still say "by the consumer" when a consumer is no longer present, I've left the issues open. As far as I can tell every other crit is addressed, and the PR can be merged after this.

**Comment by Patbox** [comment]
Not sure how I missed that one again, but now it's changed

**Pull request merged**
